src/main/java/org/mitre/synthea/world/agents/behaviors/providerfinder/ProviderFinderPreferOne.java

```diff
@@ -0,0 +1,147 @@
+package org.mitre.synthea.world.agents.behaviors.providerfinder;
+
+import java.util.List;
+
+import org.mitre.synthea.helpers.Config;
+import org.mitre.synthea.world.agents.Person;
+import java.util.Map;
+import org.mitre.synthea.world.agents.Provider;
+import org.mitre.synthea.world.concepts.HealthRecord.EncounterType;
+
+/**
+ * Finder that prioritizes a single provider based on NPI specified in configuration.
+ * Also provides a static method to override initial person demographics based on this preference.
+ * If the preferred provider is not available or suitable, it falls back to the nearest provider.
+ */
+public class ProviderFinderPreferOne implements IProviderFinder {
+
+  private static final String PREFER_ONE_NPI = "generate.providers.prefer_one.npi";
+  private static final String PREFER_ONE_IGNORE_SUITABLE = "generate.providers.prefer_one.ignore_suitable";
+  // Fallback finder if the preferred one isn't suitable during encounter finding
+  private final IProviderFinder fallbackFinder = new ProviderFinderNearest();
+
+  public static boolean isUsingPreferredProvider() {
+    return Config.get("generate.providers.selection_behavior", "nearest").equals(Provider.PREFER_ONE);
+  }
+
+  public static boolean isIgnoringSuitable() {
+    return Boolean.valueOf(Config.get(PREFER_ONE_IGNORE_SUITABLE, "false"));
+  }
+
+  public static String getPreferredNPI() {
+    return Config.get(PREFER_ONE_NPI, null);
+  }
+
+  public static Provider getPreferredProvider() {
+
+    if (!isUsingPreferredProvider()) return null;
+
+
+    String preferredNpi = getPreferredNPI();
+    if (preferredNpi == null || preferredNpi.isEmpty()) {
+      System.err.println("WARNING: generate.providers.selection_behavior=PreferOne but " + PREFER_ONE_NPI + " is not set. Using demographic location.");
+      return null; // NPI not configured, do nothing.
+    }
+
+    Provider preferredProvider = null;
+    // Find the preferred provider by NPI from the loaded list
+    // Note: This assumes the provider is within the initially loaded set.
+    for (Provider p : Provider.getProviderList()) {
+      if (preferredNpi.equals(p.npi)) {
+        preferredProvider = p;
+        break;
+      }
+    }
```

Comment:
You probably only want to perform this search once, ever, per run... rather than every time `find` is called... which can be often.